src/js/controller/controller.js

```diff
@@ -1025,15 +1040,29 @@ Object.assign(Controller.prototype, {
         };
 
         // View passthroughs
-        this.resize = _view.resize;
-        this.getSafeRegion = _view.getSafeRegion;
-        this.setCaptions = _view.setCaptions;
+        if (__HEADLESS__) {
+            // These should be overridden by the app using the headless player
+            this.resize = this.setCaptions = function() {};
+            this.getSafeRegion = () => ({
```

Comment:
Ad plugin dependency on `getSafeRegion` would be an exception if there still is a dependency there. The solution would be access the private API on setup and override the method.